subtests/docker_cli/deferred_deletion/deferred_deletion.py

```diff
@@ -107,55 +111,17 @@ def _start_idle_container(self):
         c_name = self.stuff['dc'].get_unique_name()
 
         fin = DockerImage.full_name_from_defaults(self.config)
+        self.stuff['trigger_file'] = trigger_file = 'DELETE-ME'
         subargs = ['--rm', '--name=' + c_name, fin,
-                   'bash -c "echo READY;touch /DELETE-ME;'
-                   ' while [ -e /DELETE-ME ]; do sleep 0.1; done"']
+                   'bash -c "echo READY;touch /%s;'
+                   ' while [ -e /%s ]; do sleep 0.1; done"'
+                   % (trigger_file, trigger_file)]
         dkrcmd = AsyncDockerCmd(self, 'run', subargs)
         dkrcmd.execute()
         dkrcmd.wait_for_ready(c_name)
         self.stuff['container_name'] = c_name
 
-    def _stop_idle_container(self):
-        """
-        Stop our idle container, by removing its semaphore file.
-        Wait until container is truly gone, as reported by docker ps -a
-        """
-        os.unlink('DELETE-ME')
-
-        dc = self.stuff['dc']
-        c_name = self.stuff['container_name']
-        c_gone = lambda: dc.list_containers_with_name(c_name) == []
-        self.failif(not utils.wait_for(c_gone, timeout=5),
-                    "Timed out waiting for container to exit")
-        del self.stuff['container_name']
-
-    @property
-    def _container_mountpoint(self):
-        """
-        Given a container name, return its host-accessible root filesystem
-        """
-        dc = self.stuff['dc']
-        c_name = self.stuff['container_name']
-        inspect = dc.json_by_name(c_name)
-        d_name = inspect[0]['GraphDriver']['Data']['DeviceName']
-
-        base_path = utils.run('findmnt --noheadings --output TARGET'
-                              ' --source /dev/mapper/%s' % d_name)
-        mountpoint = os.path.join(base_path.stdout.strip(), 'rootfs')
-        self.logdebug('host mount point = %s', mountpoint)
-        return mountpoint
-
-    @staticmethod
-    def _defer_count():
-        """
-        Returns the integer count of deferred deletions, as obtained
-        from docker info. We assume DockerInfo() is never cached.
-        """
-        return int(DockerInfo().get('storage_driver',
-                                    'deferred_deleted_device_count'))
-
     def cleanup(self):
         super(deferred_deletion, self).cleanup()
-        os.chdir('/')
         if 'container_name' in self.stuff:
-            self.stuff['dc'].clean_all(self.stuff['container_name'])
+            self.stuff['dc'].clean_all([self.stuff['container_name']])
```

Comment:
Sorry for the confusion.  the ``self.config['remove_after_test']`` support is a convention followed by every other test.  It's there to support runs of multiple tests over and over to check for leaks and to assist in debugging runs of an individual tests where you need to inspect the resulting container/image.  Esp. e.g. in this case, if your script doesn't terminate properly or you need to get into the container to see what's happening.

``container_name`` can be in ``stuff`` w/o ``dc`` due to future maintainer being a human, changing something, and forgetting to fixup ``cleanup()``.

This way is more forgiving of mistakes, but I'm also not going to make a fuss about it.  Checking for ``dc`` is really the only critical thing.